### DISCUSSION TITLE
Apply fluent theme inside a React Portal that's in a new Window

### DISCUSSION THREAD

**Discussion by JamesBurnside** [microsoft/fluentui#27004] [Q&A]
I'm trying to render some fluent components in a separate window, but the CSS from Fluent's ThemeProvider does not transfer over.
Is there a way to get the CSS in the theme provider to instead be in the react portal document/keep the react portal document CSS in sync with the primary document?

Example code:
```tsx
import { PrimaryButton, ThemeProvider } from '@fluentui/react';
import React, { useMemo } from 'react';
import { Portal } from '@fluentui/react-portal';

const App= (): JSX.Element => {
  return (
    <>
      This is the primary document. Rendering test text so this is not blank.
      <NewWindow>
        <ThemeProvider>
          <PrimaryButton>Test - this should have fluent styles applied but doesn't</PrimaryButton>
        </ThemeProvider>
      </NewWindow>
    </>
  );
};

const NewWindow = (props: { children: React.ReactChild; close?: () => void }): JSX.Element => {
  const newWindow = useMemo(
    () =>
      window.open(
        'about:blank',
        'newWin',
        `width=400,height=300,left=${window.screen.availWidth / 2 - 200},top=${window.screen.availHeight / 2 - 150}`
      ),
    []
  );

  if (!newWindow || !props.children) {
    console.log('Could not create new window', newWindow, props.children);
    return <>Err: Could not render in new window</>;
  }

  return (
    <Portal mountNode={newWindow.document.body}>
      {props.children}
    </Portal>
  );
};

export default App;
```

**Comment by wbarrantes** [comment]
Hey @JamesBurnside

We were facing a similar issue in OWA but using v9 components.

With [this hook](https://github.com/microsoft/fluentui/issues/23626#issuecomment-1162255474) provided by @layershifter as a workaround does the work. If you are using it for a main window, it should be also used for a child window.

Maybe this could help.

**Reply by layershifter** [reply]
With v9, it's a different thing. You can use `FluentProvider` & `RendererProvider` there to configure rendering: https://react.fluentui.dev/?path=/docs/concepts-developer-advanced-configuration--page#child-window-rendering

With v8 AFAIK there is no a similar way to do that (https://github.com/microsoft/fluentui/issues/8192).
